source/shared/core_conn.cpp

```diff
@@ -150,21 +150,55 @@ sqlsrv_conn* core_sqlsrv_connect( _In_ sqlsrv_context& henv_cp, _In_ sqlsrv_cont
     
     build_connection_string_and_set_conn_attr( conn, server, uid, pwd, options_ht, valid_conn_opts, driver, conn_str TSRMLS_CC );
     
-    bool is_missing_driver = false;
+    bool is_missing_driver = true;
 
     if ( conn->is_driver_set ) {
         r = core_odbc_connect( conn, conn_str, is_missing_driver, is_pooled );
     }
     else if ( conn->ce_option.enabled ) {
+#ifndef _WIN32
+        // In non-Windows environment, because different unixODBC versions return different error states,
+        // if it fails to connect it is unreliable to check for a certain sql state error. Thus, 
+        // we will ignore the returned value of 'is_missing_driver' and simply throw an exception.
```

Comment:
it would be cleaner to put this logic inside core_odbc_connect so it will set is_missing_driver correct then we shouldn't need to ignore it.